labgrid/util/helper.py

```diff
@@ -30,20 +32,30 @@ class ProcessWrapper:
     def check_output(self, command):
         """Run a command and supply the output to callback functions"""
         res = []
-        process = subprocess.Popen(command, stderr=subprocess.STDOUT,
-                                   stdout=subprocess.PIPE, bufsize=0)
+        mfd, sfd = pty.openpty()
+        for fd in [mfd, sfd]:
```

Comment:
@jluebbe mentioned that here the fd is also manipulated for the child. This could result in different behaviour for the spawned program, I'll reinvestigate again if the slave fd also needs to be set to `O_NONBLOCK`.

Reply:
This works fine without touching `sfd`, I'll amend the Commit.